FRENCH/src/ch12-03-improving-error-handling-and-modularity.md

```diff
@@ -0,0 +1,1308 @@
+appeler à la place le `Config::new`. Nous avons changé le nom de
+`interpreter_config` par `new` et nous l'avons déplacé dans un bloc `impl`,
+ce qui relie la fonction `new` à `Config`. Essayez à nouveau de compiler ce
+code pour vous assurer qu'il fonctionne.
+
+<!--
+### Fixing the Error Handling
+-->
+
+### Corriger la gestion des erreurs
+
+<!--
+Now we’ll work on fixing our error handling. Recall that attempting to access
+the values in the `args` vector at index 1 or index 2 will cause the program to
+panic if the vector contains fewer than three items. Try running the program
+without any arguments; it will look like this:
+-->
+
+Maintenant, nous allons nous pencher sur la correction de la gestion des
+erreurs. Rappellez-vous que la tentative d'accéder aux valeurs dans le vecteur
+`args` aux indices 1 ou 2 va faire paniquer le programme si le vecteur contient
+moins de trois éléments. Essayez de lancer le programme sans aucun argument ;
+cela donnera quelque chose comme ceci :
+
+<!--
+```console
+{{#include ../listings-sources/ch12-an-io-project/listing-12-07/output.txt}}
+```
+-->
+
+```console
+{{#include ../listings/ch12-an-io-project/listing-12-07/output.txt}}
+```
+
+<!--
+The line `index out of bounds: the len is 1 but the index is 1` is an error
+message intended for programmers. It won’t help our end users understand what
+happened and what they should do instead. Let’s fix that now.
+-->
+
+La ligne `index out of bounds: the len is 1 but the index is 1` est un
+message d'erreur destiné aux développeurs. Il n'aidera pas nos utilisateurs
+finaux à comprendre ce qu'il s'est passé et ce qu'ils devraient faire à la
+place. Corrigeons cela dès maintenant.
+
+<!--
+#### Improving the Error Message
+-->
+
+#### Améliorer le message d'erreur
+
+<!--
+In Listing 12-8, we add a check in the `new` function that will verify that the
+slice is long enough before accessing index 1 and 2. If the slice isn’t long
+enough, the program panics and displays a better error message than the `index
+out of bounds` message.
+-->
+
+Dans l'encart 12-8, nous ajoutons une vérification dans la fonction `new`, qui
+va vérifier que le slice est suffisamment grand avant d'accéder aux indices 1
+et 2. Si le slice n'est pas suffisamment grand, le programme va paniquer et
+afficher un meilleur message d'erreur que le message `index out of bounds`.
+
+<!--
+<span class="filename">Filename: src/main.rs</span>
+-->
+
+<span class="filename">Fichier : src/main.rs</span>
+
+<!--
+```rust,ignore
+{{#rustdoc_include ../listings-sources/ch12-an-io-project/listing-12-08/src/main.rs:here}}
+```
+-->
+
+```rust,ignore
+{{#rustdoc_include ../listings/ch12-an-io-project/listing-12-08/src/main.rs:here}}
+```
+
+```rust,ignore
+// -- partie masquée ici --
+fn new(args: &[String]) -> Config {
+    if args.len() < 3 {
+        panic!("il n'y a pas assez d'arguments");
+    }
+    // -- partie masquée ici --
+```
+
```

Comment:
```suggestion
```

linked to @yozhgoor suggestion in https://github.com/Jimskapt/rust-book-fr/pull/137#r672536850